packages/editor/src/editor/sidebar/fields/datatable-columns/ColumnsCheckboxField.tsx

```diff
@@ -15,12 +16,10 @@ export const ColumnsCheckboxField = ({ onChange }: ColumnsCheckboxFieldProps) =>
   const handleCheckboxChange = (e: boolean | 'indeterminate', column: CheckboxColumn) => {
     if (e === true) {
       const newColumn: DataTableColumn = {
-        config: {
-          header: column.config.header,
-          value: column.config.value,
-          sortable: column.config.sortable,
-          filterable: column.config.filterable
-        },
+        config: DataTableColumnComponent.create({
+          label: column.config.header,
+          value: column.config.value
+        }),
```

Comment:
if you use it, make sure that the create method is implemented the right way (that the label and value are correctly applied)